node/src/accountant/scanners/mod.rs

```diff
@@ -854,24 +866,31 @@ impl Scanner<RetrieveTransactions, ReceivedPayments> for ReceivableScanner {
     }
 
     fn finish_scan(&mut self, msg: ReceivedPayments, logger: &Logger) -> Option<NodeToUiMessage> {
-        if msg.payments.is_empty() {
-            info!(
-                logger,
-                "No newly received payments were detected during the scanning process."
-            );
-
-            match self
-                .persistent_configuration
-                .set_start_block(msg.new_start_block)
-            {
-                Ok(()) => debug!(logger, "Start block updated to {}", msg.new_start_block),
-                Err(e) => panic!(
-                    "Attempt to set new start block to {} failed due to: {:?}",
-                    msg.new_start_block, e
-                ),
+        match msg.scan_result {
+            Ok(payments_and_start_block) => {
+                self.handle_new_received_payments(&payments_and_start_block, msg.timestamp, logger);
             }
-        } else {
-            self.handle_new_received_payments(&msg, logger)
+            Err(e) => match e {
+                ReceivedPaymentsError::ExceededBlockScanLimit(max_block_count) => {
+                    debug!(logger, "Writing max_block_count({})", max_block_count);
+                    self.persistent_configuration
+                        .set_max_block_count(Some(max_block_count))
+                        .map_or_else(|_| {
+                            warning!(logger, "{:?} update max_block_count to {}. Scheduling next scan with that limit.", e, max_block_count);
+                        },
+                        |e| {
+                            warning!(logger, "Writing max_block_count failed: {:?}", e);
```

Comment:
This violates the rule that has been agreed on by the team (an excuse could be that this happened so long ago) which is that any error returned from the sqlite database manager will be treated by an instant panic. In other words, we decided for no-tolerance policy in the pursue of preserving the data in the database in as good state as possible, because there must have been something serious that happened in there and continuous operations could lead to just more and more corruption with us looking away.

Reply:
**URGENCY STATUS: I'D LIKE TO SEE THE POLICY BE KEPT. PANIC SHOULD HAPPEN**